### PR TITLE
Support environment variables for bots

### PR DESCRIPTION
**Pull request by rbro112**
My bot requires API token, figured it would be easier to keep in my `ryan.env` file (gitignored - checked in here so Mitch can add my API key and leave there). Then passes contents of env file to docker container.

Tested locally and my bot seemed to successfully pick, indicating it was likely picked up, but couldn't see logs to confirm.

### CODE REVIEW

File: pkg/common/agent_grpc.pb.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go-grpc. DO NOT EDIT.
 // versions:
 // - protoc-gen-go-grpc v1.5.1
-// - protoc             v4.25.2
+// - protoc             v5.28.1
```
Comment (rbro112) [inline pkg/common/agent_grpc.pb.go:4]: Can regenerate with lower if needed, seems my local is higher?

Comment (mitchwebster) [reply pkg/common/agent_grpc.pb.go:4]: Nice this Pr looks good, and ya if you want to just skip checking in the generated files you can also do that and I can regenerate them when we finish all the proto updates tomorrow

**Comment by justbuchanan** [comment]
FYI, I just pushed a change that saves log files for each bot invocation. If you run the engine with `--enable_verbose_logging`, you'll see something like `Wrote bot logs to "/tmp/draft.1971650796.stdout", "/tmp/draft.536838763.stderr"` in the output, then you can check the stdout file to see the results of any `print()` statements from your bot.

**Comment by rbro112** [comment]
Can't fully test as all containers keep hitting this error:
```
Traceback (most recent call last):
  File "/app/server.py", line 8, in <module>
    from agent_pb2_grpc import AgentServiceServicer, add_AgentServiceServicer_to_server
  File "/app/agent_pb2_grpc.py", line 6, in <module>
    import agent_pb2 as agent__pb2
  File "/app/agent_pb2.py", line 27, in <module>
    DESCRIPTOR = ...
TypeError: Couldn't build proto file into descriptor pool: duplicate file name agent.proto
```

I think this is likely just my environment/setup causing issues, but don't wanna take everything down. Will try to test again after work

**Comment by mitchwebster** [comment]
Ah ya this can happen when the pygrpc server (it uses a pinned version of the blitzenv library to ensure it updates) doesn't have the same version of the proto as the library. Should be fine to bypass this. Something we could improve in the devloop if we could get docker to pick up the most recent blitzenv each time.

**Review by mitchwebster — APPROVED** [review]

**Pull request merged**